SDWebImage/SDImageIOCoder.m

```diff
@@ -268,32 +261,32 @@ + (BOOL)canDecodeFromHEIFFormat {
     static BOOL canDecode = NO;
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
-        CFStringRef imageUTType = [NSData sd_UTTypeFromImageFormat:SDImageFormatHEIF];
-        NSArray *imageUTTypes = (__bridge_transfer NSArray *)CGImageSourceCopyTypeIdentifiers();
-        if ([imageUTTypes containsObject:(__bridge NSString *)(imageUTType)]) {
-            canDecode = YES;
-        }
+        canDecode = [self canDecodeFromFormat:SDImageFormatHEIF];
     });
     return canDecode;
 }
 
++ (BOOL)canEncodeToFormat:(SDImageFormat)format {
+    NSMutableData *imageData = [NSMutableData data];
+    CFStringRef imageUTType = [NSData sd_UTTypeFromImageFormat:format];
+    
+    // Create an image destination.
+    CGImageDestinationRef imageDestination = CGImageDestinationCreateWithData((__bridge CFMutableDataRef)imageData, imageUTType, 1, NULL);
+    if (!imageDestination) {
```

Comment:
Actually, this logic, it's a little not common. I can explain the detail.

For example, if the UTI type, is GIF (`kUTTypeGIF`). If you call `CGImageDestinationCreateWithData` with `count` arg to 1. This API will return NULL. Because Apple assume all GIF images, are animated. (Which means, you can not create a static GIF using Image/IO).

But anyway, current logic for HEIF/HEIC it's OK. But I suggest to add some  comment here (No code change needed). Like this case I talke about.